airflow/example_dags/example_xcom.py

```diff
@@ -78,6 +104,14 @@ def puller(**kwargs):
     pull = PythonOperator(
         task_id='puller',
         python_callable=puller,
+        op_kwargs={
+            'pulled_value_1': push1.output['value from pusher 1'],
+            'pulled_value_3': push2.output,
+        },
```

Comment:
I think that we shouldn't use PythonOperator at all for this example if we want to use the TaskFlow API because TaskFlowAPI is an abstraction of the PythonOperator. Using both the TaskFlow API and PythonOperator in the same example doesn't look great to me.

Reply:
```python

from airflow import DAG
from airflow.utils.dates import days_ago

value_1 = [1, 2, 3]
value_2 = {'a': 'b'}

dag = DAG(
    'example_xcom',
    schedule_interval="@once",
    start_date=days_ago(2),
    tags=['example'],
)

@dag.task()
def push():
    """Pushes an XCom without a specific target"""
    return value_1


@dag.task()
def push_by_returning():
    """Pushes an XCom without a specific target, just by returning it"""
    return value_2


@dag.task()
def puller(data1, data2):
    """Pull all previously pushed XComs and check if the pushed values match the pulled values."""

    # get value_1
    pulled_value_1 = data1
    if pulled_value_1 != value_1:
        raise ValueError(f'The two values differ {pulled_value_1} and {value_1}')

    # get value_2
    pulled_value_2 = data2
    if pulled_value_2 != value_2:
        raise ValueError(f'The two values differ {pulled_value_2} and {value_2}')

    # get both value_1 and value_2
    pulled_value_1, pulled_value_2 = data1, data2
    if pulled_value_1 != value_1:
        raise ValueError(f'The two values differ {pulled_value_1} and {value_1}')
    if pulled_value_2 != value_2:
        raise ValueError(f'The two values differ {pulled_value_2} and {value_2}')


puller(push(), push_by_returning())

```

Reply:
If we want to push with a specific ID:

```python
from airflow import DAG
from airflow.utils.dates import days_ago

value_1 = [1, 2, 3]
value_2 = {'a': 'b'}

dag = DAG(
    'example_xcom',
    schedule_interval="@once",
    start_date=days_ago(2),
    tags=['example'],
)

@dag.task()
def push(**kwargs):
    """Pushes an XCom without a specific target"""
    kwargs['ti'].xcom_push(key='value from pusher 1', value=value_1)


@dag.task()
def push_by_returning():
    """Pushes an XCom without a specific target, just by returning it"""
    return value_2


@dag.task()
def puller(data1, data2, **kwargs):
    """Pull all previously pushed XComs and check if the pushed values match the pulled values."""
    ti = kwargs['ti']
    # get value_1
    pulled_value_1 = ti.xcom_pull(key=None, task_ids='push')
    if pulled_value_1 != value_1:
        raise ValueError(f'The two values differ {pulled_value_1} and {value_1}')

    # get value_2
    pulled_value_2 = data2
    if pulled_value_2 != value_2:
        raise ValueError(f'The two values differ {pulled_value_2} and {value_2}')

    # get both value_1 and value_2
    pulled_value_1, pulled_value_2 = ti.xcom_pull(key=None, task_ids=['push', 'push_by_returning'])
    if pulled_value_1 != value_1:
        raise ValueError(f'The two values differ {pulled_value_1} and {value_1}')
    if pulled_value_2 != value_2:
        raise ValueError(f'The two values differ {pulled_value_2} and {value_2}')


puller(push(), push_by_returning())
```

Reply:
I think this is simple, easy to understand, and demonstrated the power of Taskflow API

Reply:
@ephraimbuddy @ashb @uranusjr What do you think about creating a new example DAG that strictly shows `XCom` push/pull with the Taskflow API? I agree with Ephraim and feel we should have an example for Taskflow API here -- albeit the original thought was to demonstrate using the `get_current_context()` method -- especially if see (or want) the community to adopt this style. (@ephraimbuddy @uranusjr love to get your thoughts on showcasing that method as well).

Reply:
What would the existing examples look like if we add the new example? I think it’d be a good idea to have separate examples demostrating the Taskflow API and the “traditional” context manager or `Operator(dag=dag)` approach, since they do have slightly different best practices.

Reply:
For me, I think we shouldn't show any example with PythonOperator. We should instead use TaskFlow API. However, wherever we have BashOperator and the likes, we should use context manager or `Operator(dag=dag)`. TaskFlow can be used inside context manager, for cases where we have a combination of PythonOperator and Other Operators:

```python
with dag:
    task1 = BashOperator(task_id='myid', bash_command='sleep 1')
    
    @task
    def mypythontask(**kwargs):
        return 'xcom'

task1 >> mypythontask()
```
I feel we should stop using PythonOperator in examples.

Reply:
I do have #9415 assigned to me as well to update all of the example DAGs to use the Taskflow API so eventually we'll be getting to a point where `PythonOperator` and `PythonVirtualEnvOperator` (and perhaps `TaskGroup` context manager if it makes sense in the example) are not used in the examples or documentation.  

Initially the thought was to tackle #10285 first (this PR and others to use the `.output` property and refactoring the `default_args` pattern mainly) then go back to tackle #9415 since I was already deep into updating 86 files encompassing #10285.  

Seems like it makes sense to remove the Taskflow API example in `example_xcom.py` and then continue/revisit this discussion in a later PR.  WDYT?

Reply:
Sounds good...

Reply:
Reverted Taskflow API and `get_current_context()` changes in the "example_xcom" DAG.